src/earthkit/data/readers/geotiff.py

```diff
@@ -0,0 +1,219 @@
+# (C) Copyright 2024 ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+#
+
+import mimetypes
+from functools import cached_property
+
+import numpy as np
+
+from ..core.fieldlist import Field
+from ..core.fieldlist import FieldList
+from ..core.geography import Geography
+from ..core.metadata import RawMetadata
+from ..utils.bbox import BoundingBox
+from ..utils.projections import Projection
+from . import Reader
+
+
+class GeoTIFFGeography(Geography):
+
+    def __init__(self, ds):
+        self._ds = ds
+        self.x_dim = ds.rio.x_dim
+        self.y_dim = ds.rio.y_dim
+
+    def _latlon_coords(self):
+        from pyproj import Transformer
+
+        return Transformer.from_crs(self._ds.rio.crs, "EPSG:4326", always_xy=True).transform(
+            *self._xy_coords()
+        )
+
+    def latitudes(self, dtype=None):
+        return self._latlon_coords()[0]
+
+    def longitudes(self, dtype=None):
+        return self._latlon_coords()[1]
+
+    def _xy_coords(self):
+        x = self._ds.coords[self.x_dim].values
+        y = self._ds.coords[self.y_dim].values
+        return np.meshgrid(x, y)
+
+    def x(self, dtype=None):
+        return self._xy_coords()[0]
+
+    def y(self, dtype=None):
+        return self._xy_coords()[1]
+
+    def shape(self):
+        return self._ds.rio.height, self._ds.rio.width
+
+    def _unique_grid_id(self):
+        return (*self.shape(), self._ds.rio.crs.to_wkt())
+
+    def projection(self):
+        return Projection.from_cf_grid_mapping(**self._grid_mapping)
+
+    @property
+    def _grid_mapping(self):
+        if self._ds.rio.grid_mapping == "spatial_ref":
+            return self._ds.coords["spatial_ref"].attrs
+        self._not_implemented()
+
+    def bounding_box(self):
+        left, bottom, right, top = self._ds.rio.transform_bounds("EPSG:4326")
+        return BoundingBox(north=top, west=left, south=bottom, east=right)
+
+    def gridspec(self):
+        self._not_implemented()
+
+    def resolution(self):
```

Comment:
This is a tricky method. For GRIB, returns a single value, and asserts if the resolution is different in x and y directions.